weather-app/app/components/Sidebar.tsx

```diff
@@ -0,0 +1,65 @@
+"use client";
+
+import Image from "next/image";
+import Link from "next/link";
+import React, { useState } from "react";
+import styles from "./Sidebar.module.scss";
+
+const Sidebar = () => {
+  const [isWeatherActive, setIsWeatherActive] = useState<boolean>(true);
+  const [isFavoriteListActive, setIsFavoriteListActive] =
+    useState<boolean>(false);
+
+  const weatherIconHandler = () => {
+    if (!isWeatherActive) {
+      setIsWeatherActive(true);
+      setIsFavoriteListActive(false);
+    }
+  };
+
+  const favoriteListIconHandler = () => {
+    if (!isFavoriteListActive) {
+      setIsWeatherActive(false);
+      setIsFavoriteListActive(true);
+    }
+  };
+
```

Comment:
You can remove from Line 9 `const [isWeatherActive, setIsWeatherActive] = useState<boolean>(true);` to Line 25 `};`. Instead, you can use this code `const pathname = usePathname();` to use the current path.